src/Manager/Environment/effects/removeEnvironment.ts

```diff
@@ -23,27 +23,42 @@ import {
 } from '../environmentReducer';
 import { removeDir } from './removeDir';
 
-const removeLegacyEnvironment = (dispatch: Dispatch, toolchainDir: string) =>
-    removeDir(dispatch, path.dirname(toolchainDir));
+const removeLegacyEnvironment = (toolchainDir: string) =>
+    removeDir(path.dirname(toolchainDir));
 
-const removeNrfutilEnvironment = (dispatch: Dispatch, version: string) =>
-    Promise.all([
-        removeToolchain(version),
-        removeDir(dispatch, sdkPath(version)),
-    ]);
+const removeNrfutilEnvironment = (version: string) => {
+    removeDir(sdkPath(version));
+    try {
+        removeToolchain(version);
+    } catch (err) {
+        throw new Error(
+            `Failed to remove ${sdkPath(version)}, ${err}. ` +
+                `Please remove the installation for ${version} manually as it is broken.`
+        );
+    }
+};
 
 export const removeEnvironment =
-    (environment: Environment) => async (dispatch: Dispatch) => {
+    (environment: Environment) => (dispatch: Dispatch) => {
         const { toolchainDir, version } = environment;
         logger.info(`Removing ${version} at ${toolchainDir}`);
         usageData.sendUsageData(EventAction.REMOVE_TOOLCHAIN, `${version}`);
 
         dispatch(startRemoving(version));
 
-        if (isLegacyEnvironment(version)) {
-            await removeLegacyEnvironment(dispatch, toolchainDir);
-        } else {
-            await removeNrfutilEnvironment(dispatch, version);
+        try {
+            if (isLegacyEnvironment(version)) {
+                removeLegacyEnvironment(toolchainDir);
+            } else {
+                removeNrfutilEnvironment(version);
+            }
+        } catch (err) {
+            dispatch(
+                ErrorDialogActions.showDialog(`${(err as Error).message}`)
+            );
+            usageData.sendErrorReport((err as Error).message);
+            dispatch(finishRemoving(version));
+            return;
```

Comment:
Instead of putting a `dispatch(finishRemoving(version)); return;` into the catch-block, I think it would be more readable to pull the logging and `removeEnvironmentReducer` into the try-block.

So, instead of 

```ts
…
try {
  …
} catch (err) {
  …
  dispatch(finishRemoving(version));
  return;
}
logger.info(`Finished removing ${version} at ${toolchainDir}`);
dispatch(removeEnvironmentReducer(version));

dispatch(finishRemoving(version));
```

rather:

```ts
…
try {
  …

  logger.info(`Finished removing ${version} at ${toolchainDir}`);
  dispatch(removeEnvironmentReducer(version));
} catch (err) {
  …
}
dispatch(finishRemoving(version));
```